index.d.ts

```diff
@@ -100,6 +100,8 @@ interface ClientOptions {
   maxRetries?: number;
   requestTimeout?: number;
   pingTimeout?: number;
+  responseMaxHeapPercentage?: number;
```

Comment:
This options go together like 🍪 & 🥛,

I think we should consider something like:
```
memoryCircuitBreaker?: {
  enabled: boolean | false;
  maxHeapPercentage: number | 1.0;
}
```